arch/arm/core/cortex_m/Kconfig

```diff
@@ -35,20 +35,84 @@ config ISA_THUMB2
 	  technology is featured in the  processor, and in all ARMv7
 	  architecture-based processors.
 
+config CPU_CORTEX_M_HAS_PRIMASK
+	bool
+	# Omit prompt to signify "hidden" option
+	default y
+	help
+	  This option signifies the CPU has the PRIMASK register.
+
+	  The PRIMASK register prevents activation of all exceptions
+	  with configurable priority. Always present in CPUs that
+	  implement the ARMv6-M, ARMv7-M, or ARMv8-M architectures.
+
 config CPU_CORTEX_M_HAS_BASEPRI
 	bool
 	# Omit prompt to signify "hidden" option
+	depends on ARMV7_M_ARMV8_M_MAINLINE
 	default n
 	help
 	  This option signifies the CPU has the BASEPRI register.
 
+	  The BASEPRI register defines the minimum priority for
+	  exception processing. When BASEPRI is set to a nonzero
+	  value, it prevents the activation of all exceptions with
+	  the same or lower priority level as the BASEPRI value.
+	  Always present in CPUs that implement the ARMv7-M or
+	  ARM8-M Mainline architectures.
+
+config CPU_CORTEX_M_HAS_FAULTMASK
```

Comment:
what code is using CPU_CORTEX_M_HAS_FAULTMASK?

Reply:
@galak  WIll re-work this PR (overlaps partially with other PRs); ignore at the moment.